honeybee_schema/radiance/material.py

```diff
@@ -0,0 +1,280 @@
+
+    g_reflectance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the green channel reflectance '
+                    '(default: 0).'
+    )
+
+    b_reflectance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the blue channel reflectance '
+                    '(default: 0).'
+    )
+
+    specularity: float = Field(
+        default=0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the fraction of specularity. '
+                    'Specularity fractions greater than 0.1 are not realistic. '
+                    '(default: 0).'
+    )
+
+    roughness: float = Field(
+        default=0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the roughness, specified as the '
+                    'rms slope of surface facets. Roughness greater than 0.2 are '
+                    'not realistic (default: 0).'
+    )
+
+
+class Trans(Plastic):
+    """Radiance Translucent material."""
+
+    type: constr(regex='^Trans$') = 'Trans'
+
+    transmitted_diff: float = Field(
+        default=0,
+        description='The fraction of transmitted light that is transmitted diffusely in '
+                    'a scattering fashion (default: 0).'
+    )
+
+    transmitted_spec: float = Field(
+        default=0,
+        description='The fraction of transmitted light that is not diffusely scattered '
+                    '(default: 0).'
+    )
+
+
+class Glass(ModifierBase):
+    """Radiance glass material."""
+
+    type: constr(regex='^Glass$') = 'Glass'
+
+    r_transmissivity: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the red channel transmissivity '
+                    '(default: 0).'
+    )
+
+    g_transmissivity: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the green channel transmissivity '
+                    '(default: 0).'
+    )
+
+    b_transmissivity: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the blue channel transmissivity '
+                    '(default: 0).'
+    )
+
+    refraction_index: float = Field(
+        default=1.52,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the index of refraction '
+                    '(default: 1.52).'
+    )
+
+
+class BSDF(ModifierBase):
+    """Radiance BSDF (Bidirectional Scattering Distribution Function) material."""
+
+    type: constr(regex='^BSDF$') = 'BSDF'
+
+    up_orientation: List[float] = Field(
+        default=(0.01, 0.01, 1.00),
+        description='Vector as sequence that sets the hemisphere that the BSDF material '
+                    'faces. (default: (0.01, 0.01, 1.00).'
+    )
+
+    thickness: float = Field(
+        default=0,
```

Comment:
missing check for number to be positive.

Reply:
@mostaphaRoudsari I believe thickness can be negative to indicate the geometry is projected behind the BSDF surface (from the honeybee_radiance docstrings).

Reply:
Yes. You're right. My bad. Disregard this comment.